## Load the data

In [ ]:
import pandas as pd
df = pd.read_csv('/content/KAG_Conversion_Data.csv')
df

ad_id  xyz_campaign_id  fb_campaign_id    age gender  interest  \
0      708746              916          103916  30-34      M        15   
1      708749              916          103917  30-34      M        16   
2      708771              916          103920  30-34      M        20   
3      708815              916          103928  30-34      M        28   
4      708818              916          103928  30-34      M        28   
...       ...              ...             ...    ...    ...       ...   
1138  1314410             1178          179977  45-49      F       109   
1139  1314411             1178          179978  45-49      F       110   
1140  1314412             1178          179979  45-49      F       111   
1141  1314414             1178          179981  45-49      F       113   
1142  1314415             1178          179982  45-49      F       114   

      Impressions  Clicks       Spent  Total_Conversion  Approved_Conversion  
0            7350       1    1.430000                 2                    1  
1           17861       2    1.820000                 2                    0  
2             693       0    0.000000                 1                    0  
3            4259       1    1.250000                 1                    0  
4            4133       1    1.290000                 1                    1  
...           ...     ...         ...               ...                  ...  
1138      1129773     252  358.189997                13                    2  
1139       637549     120  173.880003                 3                    0  
1140       151531      28   40.289999                 2                    0  
1141       790253     135  198.710001                 8                    2  
1142       513161     114  165.609999                 5                    2  

[1143 rows x 11 columns]

In [ ]:
import numpy as np
data_file = '/content/KAG_Conversion_Data.csv'
data = np.genfromtxt(data_file, delimiter=",", skip_header=1,dtype=str)
print("\nData: \n\n",data)


Data: 

 [['708746' '916' '103916' ... '1.429999948' '2' '1']
 ['708749' '916' '103917' ... '1.820000023' '2' '0']
 ['708771' '916' '103920' ... '0' '1' '0']
 ...
 ['1314412' '1178' '179979' ... '40.28999949' '2' '0']
 ['1314414' '1178' '179981' ... '198.7100005' '8' '2']
 ['1314415' '1178' '179982' ... '165.6099987' '5' '2']]


## Unique ad campaigns = 1143
## No. of times each campaign was run = 

In [ ]:
U_adcampaigns = data[:,1]
U_adcampaigns

array(['916', '916', '916', ..., '1178', '1178', '1178'], dtype='<U11')

In [ ]:
U_adcampaigns.shape
np.unique(U_adcampaigns)

array(['1178', '916', '936'], dtype='<U11')

In [ ]:
len(data[data[:,1]=='916'])

54

In [ ]:
len(data[data[:,1]=='936'])

464

In [ ]:
len(data[data[:,1]=='1178'])

625

In [ ]:
len(U_adcampaigns)

1143

In [ ]:
age_group= data[:,3]
age_group

array(['30-34', '30-34', '30-34', ..., '45-49', '45-49', '45-49'],
      dtype='<U11')

In [ ]:
np.unique(age_group)

array(['30-34', '35-39', '40-44', '45-49'], dtype='<U11')

What was the average, minimum and maximum amount spent on the ads?



In [25]:
spent = data[:,8]
spent

array(['1.429999948', '1.820000023', '0', ..., '40.28999949',
       '198.7100005', '165.6099987'], dtype='<U11')

In [26]:
amt_spent = spent.astype(float)
amt_spent

array([  1.42999995,   1.82000002,   0.        , ...,  40.28999949,
       198.7100005 , 165.6099987 ])

In [27]:
min_amt = amt_spent.min()
min_amt

0.0

In [28]:
max_amt = amt_spent.max()
max_amt

639.9499981

What is the id of the ad having the maximum number of clicks ?



In [34]:
click = data[:,7]

click = click.astype(float)
click

array([  1.,   2.,   0., ...,  28., 135., 114.])

What is the id of the ad having the maximum number of clicks ?



In [35]:
max_click = click.max()
max_click

421.0

In [57]:
data[:,0][click==421]

array(['1121814'], dtype='<U11')

In [59]:

#max_click_id
# max click id = 1121814

How many people bought the product after seeing the ad with most clicks? Is that the maximum number of purchases in this dataset?



In [60]:
purchase = data[:,10] #approved coversions
purchase = purchase.astype(float)
purchase

array([1., 0., 0., ..., 0., 2., 2.])

In [62]:
max_purchase = purchase.max()
max_purchase

21.0

So the ad with the most clicks didn't fetch the maximum number of purchases. Find the details of the product having maximum number of purchases

1121104

In [63]:
data[:,0][purchase==21]

array(['1121104'], dtype='<U11')

21.0